lib/src/rewrite.rs

```diff
@@ -190,6 +190,24 @@ impl<'repo> CommitRewriter<'repo> {
         }
     }
 
+    /// Update the intended new parents by replacing any occurrence of
+    /// `old_parent` by `unrewritten_parents`.
+    pub fn replace_rewritten_parent<'a>(
+        &mut self,
+        old_parent: &CommitId,
+        unrewritten_parents: impl IntoIterator<Item = &'a CommitId>,
+    ) {
+        let old_parent = &self.mut_repo.new_parents(&[old_parent.clone()])[0];
```

Comment:
Can we assume that `new_parents()` never returns more than one parent? It might be safer to loop over "new" old parents, but I'm not pretty sure.